src/lib/components/submit/SubmitAddNodeProvider.svelte

```diff
@@ -90,6 +91,61 @@
 
 			debounceSave();
 		})();
+
+	export function checkFields(): boolean {
+		const validUrlDomain = 'https://wiki.internetcomputer.org/';
+		const sha256Regex = /^[a-fA-F0-9]{64}$/;
+		const fields = [
+			nodeProviderName,
+			url,
+			urlSelfDeclaration,
+			hashSelfDeclaration,
+			urlIdentityProof,
+			hashIdentityProof,
+			nodeProviderId
+		];
+
+		if (fields.some((field) => field === '')) {
```

Comment:
fields can be undefined as well?

Reply:
Correct me if I am wrong, but I don't think so.
```
const init = () => {
		nodeProviderName = $store?.metadata?.nodeProviderName ?? '';
		url = $store?.metadata?.url ?? '';
		urlSelfDeclaration = $store?.metadata?.urlSelfDeclaration ?? '';
		urlIdentityProof = $store?.metadata?.urlIdentityProof ?? '';
		hashSelfDeclaration = $store?.metadata?.hashSelfDeclaration ?? '';
		hashIdentityProof = $store?.metadata?.hashIdentityProof ?? '';
		nodeProviderId = $store?.metadata?.nodeProviderId ?? '';
	};
```
The fields are always initialized with and empty string or what is stored in the store.

Reply:
To some extension, you don't have a guarantee that init is executed before any call to `checkFields`. Moreover, once you have extracted those checks from the component to a service (or else), then the fields can be undefined.